### ISSUE TITLE
function overloads in TypeScript not working

### ISSUE THREAD

**Issue by BetterCallSky** [swc-project/swc#357]
```ts
// src/addProp.ts

export function addProp<T, K extends string, V>(
  obj: T,
  prop: K,
  value: V
): T & { [x in K]: V };
export function addProp<T, K extends string, V>(
  prop: K,
  value: V
): (obj: T) => T & { [x in K]: V };

export function addProp(arg1: any, arg2: any, arg3?: any): any {
  if (arguments.length === 2) {
    return (object: any) => _addProp(object, arg1, arg2);
  }
  return _addProp(arg1, arg2, arg3);
}

function _addProp(obj: any, prop: string, value: any) {
  return {
    ...obj,
    [prop]: value,
  };
}
```
```
ERROR in ./src/addProp.ts 31:0
Module parse failed: Unexpected token (31:0)
You may need an appropriate loader to handle this file type.
| }
| export function addProp(obj, prop, value) 
> export function addProp(prop, value) 
| export function addProp(arg1, arg2, arg3) {
|     if (arguments.length === 2) {
 @ ./src/App.tsx 2:0-36 3:0-7
 @ ./src/index.tsx
```

```
    "@swc/core": "1.0.39",
    "swc-loader": "^0.1.4",
    "webpack": "^4.29.0",
```

```js
{
          loader: 'swc-loader',
          options: {
            jsc: {
              parser: {
                syntax: 'typescript',
                tsx: true,
              },
              transform: {
                react: {
                  pragma: 'React.createElement',
                  pragmaFrag: 'React.Fragment',
                  throwIfNamespace: true,
                  development: false,
                  useBuiltins: false,
                },
              },
            },
          },
        }
```

**Comment by BetterCallSky** [comment]
@kdy1 thanks for the fix. Will you release a new version soon?

**Comment by kdy1** [comment]
@BetterCallSky I published `@swc/core@1.0.40`.
